### PR TITLE
Migrate to GitHub Actions

### LINKED ISSUE #12: `dev` branch builds are broken
Unsure of the cause, may need an SDK update:

https://ci.appveyor.com/project/Destructurama/by-ignoring/builds/41354899

### LINKED ISSUE #21: Switch to GitHub Action
I propose to switch to GitHub Action as a more robust CI ecosystem.

### PR DESCRIPTION
**Pull request by sungam3r**
fixes #21 
fixes #12

@nblumhardt I copied already known and tested workflows from other repo and adjusted them to this one.

### CODE REVIEW

**Comment by sungam3r** [comment]
@nblumhardt I have no access to repo settings to be able finish all preparations.

**Comment by sungam3r** [comment]
TODO:
1. Change default branch to `master`. Optional - remove `dev` branch.
2. Change merge option to squash.
3. Remove appveyor webhook.

**Comment by sungam3r** [comment]
By the way, readme rendering was fixed - see https://www.nuget.org/packages/Destructurama.ByIgnoring/2.0.1-dev-00085#readme-body-tab

File: src/Directory.Build.props [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    <ImplicitUsings>enable</ImplicitUsings>
+    <SignAssembly>true</SignAssembly>
+    <AssemblyOriginatorKeyFile>..\..\assets\Destructurama.snk</AssemblyOriginatorKeyFile>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(IsPackable)' == 'true'">
+    <None Include="..\..\assets\icon.png" Pack="true" PackagePath="\" Visible="false" />
+    <None Include="..\..\README.md" Pack="true" PackagePath="\" Visible="false" />
+    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="8.0.0" PrivateAssets="All" />
+    <PackageReference Include="MinVer" Version="4.3.0" PrivateAssets="All" />
```
Comment (sungam3r) [inline src/Directory.Build.props:33]: It allows automatic versioning using tags.

**Review by nblumhardt — APPROVED** [review]

File: src/Destructurama.ByIgnoring/Destructurama.ByIgnoring.csproj [hunk 2]
```diff
@@ -1,18 +1,17 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.1;netstandard2.0;netstandard1.0</TargetFrameworks>
+    <TargetFrameworks>netstandard2.1;netstandard2.0;netstandard1.1</TargetFrameworks>
```
[the remark this answers is not in the document]
Comment (sungam3r) [reply src/Destructurama.ByIgnoring/Destructurama.ByIgnoring.csproj:4]: I expect that.

**Comment by sungam3r** [comment]
Tests passed. I'll go on changing configuration.

**Comment by sungam3r** [comment]
https://github.com/destructurama/by-ignoring/pull/22#issuecomment-1895254216 - all done except removing `dev` branch

**Pull request merged**